### DISCUSSION TITLE
cppflow tutorial

### DISCUSSION THREAD

**Discussion by flaviu22** [serizba/cppflow#144] [General]
Is there a place where I can find a deeper cppflow tutorial ?

I have read this one, very basic: https://serizba.github.io/cppflow/examples.html

But I need to convert something like this:

(python code)
```
concrete_func = model.signatures[tf.saved_model.DEFAULT_SERVING_SIGNATURE_DEF_KEY]
concrete_func.inputs[0].set_shape([1, 50, 50, 3])
```
I am pretty sure this code could be converted in c++ using cppflow, but I need more details regarding how to use cppflow methods to accomplish this. For instance, how to know what is the right cppflow `model` method to replace `model.signatures[tf.saved_model.DEFAULT_SERVING_SIGNATURE_DEF_KEY]`

Can you tell me then a good cpp tutorial ?

**Comment by flaviu22** [comment]
Is this question too general ? Please suggest then any tutorial in order to solve my target.

**Comment by dskkato** [comment]
cppflow seems to have no signature handling method, so you need to investigate node names with `saved_model_cli`. You can follow [the official document](https://www.tensorflow.org/guide/saved_model#details_of_the_savedmodel_command_line_interface) how to use the tool.

Each `name` of nodes is (basically) unique regardless of the `signature_def`, and can be used in this cppflow.

```sh
$ saved_model_cli show --dir \
/tmp/saved_model_dir --tag_set serve --signature_def serving_default
The given SavedModel SignatureDef contains the following input(s):
  inputs['x'] tensor_info:
      dtype: DT_FLOAT
      shape: (-1, 1)
      name: x:0
```

Same thing is explained cppflow document [here](https://serizba.github.io/cppflow/examples.html#multi-input-output-model)
